service/lib/dinstaller/dbus/storage/guided_proposal.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "dbus"
+require "dinstaller/storage/guided_proposal"
+
+module DInstaller
+  module DBus
+    module Storage
+      # D-Bus object to manage a storage guided proposal
+      class GuidedProposal < ::DBus::Object
+        PATH = "/org/opensuse/DInstaller/Storage/GuidedProposal1"
+        private_constant :PATH
+
+        INTERFACE = "org.opensuse.DInstaller.Storage.GuidedProposal1"
+        private_constant :INTERFACE
+
+        def initialize(logger)
+          @logger = logger
+
+          super(PATH)
+        end
+
+        dbus_interface INTERFACE do
+          dbus_reader :lvm, "b", dbus_name: "LVM"
+
+          dbus_reader :candidate_devices, "as"
+
+          dbus_method :AvailableDevices, "out devices:as" do
+            backend.available_devices
+          end
+
+          dbus_method :Calculate, "in settings:a{sv}, out result:u" do |settings|
+            backend.calculate(settings)
+
+            self.ProposalChanged
+
+            backend.proposal.failed? ? 1 : 0
+          end
+
+          dbus_signal :ProposalChanged
```

Comment:
this should not be needed. It should be done automatic by ruby-dbus. See other dbus that generate properties signal.

Reply:
Ok, let's use *PropertiesChanged*.